kratos/processes/simple_mortar_mapper_process.h

```diff
@@ -516,6 +478,543 @@ class SimpleMortarMapperProcess
+                    {   
+                        const double& value = MortarUtilities::GetAuxiliarValue<TVarType>(pnode, i_size);
+                        #pragma omp atomic
+                        residual_norm[i_size] += std::pow(value, 2);
+                    }
+                }
+            }
+            
+            // Finally we compute the residual
+            for (unsigned int i_size = 0; i_size < variable_size; ++i_size)
+            {
+                residual_norm[i_size] = std::sqrt(residual_norm[i_size])/size_system;
+                if (iteration == 0) norm_b0[i_size] = residual_norm[i_size];
+                if (residual_norm[i_size] < absolute_convergence_tolerance) is_converged[i_size] = true;
+                if (residual_norm[i_size]/norm_b0[i_size] < relative_convergence_tolerance) is_converged[i_size] = true;
+                if (iteration > 0) 
+                {   
+                    increment_residual_norm = std::abs((residual_norm[i_size] - norm_bi[i_size])/norm_bi[i_size]);
+                    if (increment_residual_norm < relative_convergence_tolerance) is_converged[i_size] = true;
+                }
+                norm_bi[i_size] = residual_norm[i_size];
+                if (mEchoLevel > 0)
+                {
+                    std::cout << "Iteration: " << iteration + 1 << "\tRESISUAL::\tABS: " << residual_norm[i_size] << "\tRELATIVE: " << residual_norm[i_size]/norm_b0[i_size] << "\tINCREMENT: " << increment_residual_norm << std::endl;
+                }
+            }
+            
+            iteration += 1;
+        }
+        
+        KRATOS_CATCH("");
+    }
+    
+    /**
+     * This method executes the mapping when a linear solver is avalaible and a system of equations can be solved
+     */
+    void ExecuteImplicitMapping()
+    {
+        KRATOS_TRY;
+
+        // Defining tolerance
+        const double relative_convergence_tolerance = mThisParameters["relative_convergence_tolerance"].GetDouble();
+        const double absolute_convergence_tolerance = mThisParameters["absolute_convergence_tolerance"].GetDouble();
+        const unsigned int max_number_iterations = mThisParameters["max_number_iterations"].GetInt();
+        unsigned int iteration = 0;
+        
+        // We set to zero the variables
+        MortarUtilities::ResetValue<TVarType, THist>(mrThisModelPart, mDestinationVariable);
+        
+        // Creating the assemble database
+        std::size_t size_system;
+        std::unordered_map<int, int> conectivity_database, inverse_conectivity_database;
+        CreateSlaveConectivityDatabase(size_system, conectivity_database, inverse_conectivity_database);
+        
+        // Defining the operators
+        const unsigned int variable_size = MortarUtilities::SizeToCompute<TDim, TVarType>();
+        std::vector<bool> is_converged(variable_size, false);
+        MatrixType A = ZeroMatrix(size_system, size_system);
```

Comment:
you cannot do this. this matrix is dense after this. this is a blocker

Reply:
OK, I will change it

Reply:
Using the Dual LM I get a diagonal matrix, so I know that everything is zero except the diagonal, a simple command to do that?